src/lightning_app/cli/lightning_cli.py

```diff
@@ -125,31 +142,85 @@ def run_app(
     _run_app(file, cloud, without_server, no_cache, name, blocking, open_ui, env)
 
 
-@main.group(hidden=True)
+def app_command():
+    """Execute a function in a running application from its name."""
+    from lightning_app.utilities.commands.base import _download_command
+
+    logger.warn("Lightning Commands are a beta feature and APIs aren't stable yet.")
+
+    debug_mode = bool(int(os.getenv("DEBUG", "0")))
+
+    parser = ArgumentParser()
+    parser.add_argument("--app_id", default=None, type=str, help="Optional argument to identify an application.")
+    hparams, argv = parser.parse_known_args()
+
+    # 1: Collect the url and comments from the running application
+    url, commands = _retrieve_application_url_and_available_commands(hparams.app_id)
+    if url is None or commands is None:
+        raise Exception("We couldn't find any matching running app.")
+
+    if not commands:
+        raise Exception("This application doesn't expose any commands yet.")
+
+    command = argv[0]
+
+    command_names = [c["command"] for c in commands]
+    if command not in command_names:
+        raise Exception(f"The provided command {command} isn't available in {command_names}")
+
+    # 2: Send the command from the user
+    command_metadata = [c for c in commands if c["command"] == command][0]
+    params = command_metadata["params"]
+
+    # 3: Prepare the arguments provided by the users.
+    # TODO: Improve what is supported there.
```

Comment:
Is this still the case? Isn’t this superseded by having parsing within the command?

Reply:
If you don't use a ClientCommand, parsing is a bit clunky as resolving typing is a bit complex.